src/serializers/fields.rs

```diff
@@ -444,8 +546,19 @@ impl TypeSerializer for GeneralFieldsSerializer {
                 let filter = self.filter.key_filter(&key, include, exclude).map_err(py_err_se_err)?;
                 if let Some((next_include, next_exclude)) = filter {
                     let output_key = infer_json_key(&key, extra).map_err(py_err_se_err)?;
-                    let s = SerializeInfer::new(&value, next_include.as_ref(), next_exclude.as_ref(), extra);
-                    map.serialize_entry(&output_key, &s)?;
+                    if extra.sort_keys {
```

Comment:
While this block is cater for extra fields (not defined in pydantic schema)
So we need to sort recursively for that as well. 
Both of them doesn't seem to use dict.rs serializer . As the objective is `to_json` and using serde package directly...

https://github.com/pydantic/pydantic-core/pull/1637/files#diff-d32436e9ac9b3e5dfaf920749269f6cff3dae3b3e030561f9b9bf50447067450R540